backend/api/Controllers/RobotController.cs

```diff
@@ -654,6 +657,99 @@ public async Task<ActionResult> ResumeMission([FromRoute] string robotId)
         return Ok(mission);
     }
 
+
+    /// <summary>
+    /// Starts a mission which drives the robot to the nearest safe position
+    /// </summary>
+    /// <remarks>
+    /// <para> This query starts a localization for a given robot </para>
+    /// </remarks>
+    [HttpPost]
+    [Route("{robotId}/{asset}/{deck}/safe-position")]
+    [Authorize(Roles = Role.User)]
+    [ProducesResponseType(typeof(Mission), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<Mission>> SafePosition(
+        [FromRoute] string robotId,
+        [FromRoute] string asset,
+        [FromRoute] string deck
+    )
+    {
+        var robot = await _robotService.ReadById(robotId);
+        if (robot == null)
+        {
+            _logger.LogWarning("Could not find robot with id={id}", robotId);
+            return NotFound("Robot not found");
+        }
+
+        var assetDeck = await _assetDeckService.ReadByAssetAndDeck(asset, deck);
+        if (assetDeck is null)
+        {
+            _logger.LogWarning("Could not find any safe positions for asset={asset} and deck={deck}", asset, deck);
+            return NotFound("No safe position found");
+        }
+
+        var closestSafePosition = ClosestSafePosition(robot.Pose, assetDeck.SafePositions);
+
+        var customTaskQuery = new CustomTaskQuery
+        {
+            RobotPose = closestSafePosition,
+            Inspections = new List<CustomInspectionQuery>(),
+            InspectionTarget = new Position(),
+            TaskOrder = 0
+        };
+        var mission = new Mission
+        {
+            Name = "Drive to Safe Position Mission",
+            Robot = robot,
+            AssetCode = "NA",
+            EchoMissionId = 0,
```

Comment:
```suggestion
            EchoMissionId = null,
```

Reply:
EchoMissionId is an int, so cannot be null

Reply:
Ah, should be nullable, especially now that we have custom missions

Reply:
Made an issue

https://github.com/equinor/flotilla/issues/794